src/main/java/salesforce/ui/pages/classic/contracts/ClassicNewContractPage.java

```diff
@@ -128,25 +108,25 @@ public void setContractTermMonths(final String newContractTermMonths) {
      * @param newOwnerExpirationNotice to be typed.
      */
     public void setOwnerExpirationNotice(final String newOwnerExpirationNotice) {
-        this.ownerExpirationNotice.click();
+        webElementAction.getElement(this.ownerExpirationNotice);
         switch (newOwnerExpirationNotice) {
             case "15 Days":
-                ownerExpirationNotice.findElement(By.xpath("//option[. = '15 Days']")).click();
+                webElementAction.clickByLocator(By.xpath("//option[. = '15 Days']"));
```

Comment:
Instead of a switch with .clickByLocator you could use the webElementAction.clickOnElement() with a general String like "//option[. = '%s']" and send the selector and newOwnerExpirationNotice as parameters.